exercise/cache/exercise.py

```diff
@@ -26,6 +28,9 @@ def _generate_data(self, exercise, data=None):
                 *self.load_args,
                 last_modified=data['last_modified'] if data else None
             )
+            
+            page.content = self.compressor(page.content.encode('utf-8'))
```

Comment:
I don't think it's a good idea to overwrite the value in the `page` object. As the object is destroyed in the end of this block, I think it would be fine to just use temporary variable, e.g., just `content = ..`.